Mathlib/Tactic/Constructor.lean

```diff
@@ -0,0 +1,21 @@
+/-
+Copyright (c) 2022 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Scott Morrison
+-/
+import Lean.Elab.Command
+import Lean.Meta.Tactic.Constructor
+
+open Lean Meta Elab Tactic
+
+/--
+`fconstructor` is like `constructor`
+(it calls `apply` using the first constructor of an inductive datatype)
+except that it does not reorder goals.
+-/
+elab "fconstructor " : tactic =>
+withMainContext do
```

Comment:
This is one of the few correct (non-)uses of `liftMetaTactic`! In this case we can't use it since `Term.synthesizeSyntheticMVarsNoPostponing` is not in `MetaM`.
```suggestion
elab "fconstructor " : tactic => withMainContext do
```
...or indent everything.